test/lib/bitkassa/payment_request_test.rb

```diff
@@ -0,0 +1,141 @@
+                       "https://www.bitkassa.nl/api/v1",
+                       body: "p=#{subject.payload}&a=#{subject.authentication}"
+    end
+
+    it "returns a Bitkassa::PaymentResponse initialized with response json" do
+      subject.responder = MockPaymentRequest
+      stubbed_request.to_return(body: "response_json")
+      subject.perform
+      MockPaymentRequest.call_registry[:from_json].must_equal "response_json"
+    end
+
+    it "does not perform when merchant_id is empty" do
+      Bitkassa.config.merchant_id = ""
+      proc { subject.perform }.must_raise Bitkassa::Exception
+    end
+
+    it "does not perform when secret_api_key is empty" do
+      Bitkassa.config.secret_api_key = ""
+      proc { subject.perform }.must_raise Bitkassa::Exception
+    end
+  end
+
+  describe "#payload" do
+    it "is base64 encoded JSON with action merchant_id currency and amount" do
+      expected = {
+        "action" => "start_payment",
+        "merchant_id" => "banketbakkerhenk",
+        "currency" => "EUR",
+        "amount" => 1337 }
+
+      decoded = Base64.decode64(subject.payload)
+      parsed = JSON.parse(decoded)
+      parsed.must_equal(expected)
+    end
+  end
+
+  describe "#json_payload" do
+    it "is JSON with action merchant_id currency and amount" do
+      expected = {
+        "action" => "start_payment",
+        "merchant_id" => "banketbakkerhenk",
+        "currency" => "EUR",
+        "amount" => 1337 }
+
+      parsed = JSON.parse(subject.json_payload)
+      parsed.must_equal(expected)
+    end
+
+    it "allows description, return_url, update_url and meta_info args" do
+      bitkassa = Bitkassa::PaymentRequest.new(
+        "EUR", 1337,
+        description: "Description",
+        return_url: "http://example.com/return",
+        update_url: "http://example.com/update",
+        meta_info: "ORDERID42")
+
+      expected = {
+          "action" => "start_payment",
+          "merchant_id" => "banketbakkerhenk",
+          "currency" => "EUR",
+          "amount" => 1337,
+          "description" => "Description",
+          "return_url" => "http://example.com/return",
+          "update_url" => "http://example.com/update",
+          "meta_info" => "ORDERID42"}
+
+      parsed = JSON.parse(bitkassa.json_payload)
+      parsed.must_equal(expected)
+    end
+  end
+
+  describe "#authentication" do
+    before do
+      Bitkassa.config.secret_api_key = "SECRET"
+    end
+
+    # a = sha256( secret API key + json data + unixtime ) + unixtime
+    it "is sha256 hash of api-key, payload and current unix time" do
+      now = 42
+      Time.stub :now, Time.at(now) do
+        message = "SECRET#{subject.json_payload}#{now.to_s}"
+        expected = Digest::SHA256.hexdigest(message)
+
+        hash = subject.authentication[0...64]
+        hash.must_equal expected
+      end
+    end
+
+    it "adds current unix time to hash" do
+      now = 42
+
+      Time.stub :now, Time.at(now) do
```

Comment:
Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.at.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.